web/src/engine/main/src/hardKeyboard.ts

```diff
@@ -39,7 +39,9 @@ export function processForMnemonicsAndLegacy(s: KeyEvent, activeKeyboard: Keyboa
     //
     // Third:  DO, however, track direct presses of any main modifier key.  The OSK should
     // reflect the current modifier state even for legacy keyboards.
-    if(!activeKeyboard.definesPositionalOrMnemonic && !(s.Lmodifiers & 0x60) && !s.isModifier) {
+    if(!activeKeyboard.definesPositionalOrMnemonic &&
+       !(s.Lmodifiers & Codes.modifierBitmasks.NON_LEGACY) &&
```

Comment:
This change isn't strictly necessary; it just adds a bit more robustness to the handling.  They're shouldn't be any 1.0 keyboards with chirality handling enabled, which would be necessary to meet this condition.

Robustness:  should the `LisVirtualKey` bit flag ever be incorrect for a chiral keystroke, the chiral modifiers would be enough to continue failing the conditional.